processors/src/VertexAnaProcessor.cxx

```diff
@@ -283,8 +310,9 @@ bool VertexAnaProcessor::process(IEvent* ievent) {
             }
             
             //Add the momenta to the tracks
-            //ele_trk_gbl->setMomentum(ele->getMomentum()[0],ele->getMomentum()[1],ele->getMomentum()[2]);
-            //pos_trk_gbl->setMomentum(pos->getMomentum()[0],pos->getMomentum()[1],pos->getMomentum()[2]);
+            ele_trk_gbl->setMomentum(ele->getMomentum()[0],ele->getMomentum()[1],ele->getMomentum()[2]);
+            pos_trk_gbl->setMomentum(pos->getMomentum()[0],pos->getMomentum()[1],pos->getMomentum()[2]);
+            TVector3 recEleP(ele->getMomentum()[0],ele->getMomentum()[1],ele->getMomentum()[2]);
```

Comment:
Strange because I did run from reco LCIO tuples and I needed this for it to work. Perhaps it is correctly filled during ntuplization? I could check.

Reply:
It's a corner case. It's happen in this situations
- One runs reconstruction out of reco LCIO
- Erases FinalStateParticles in the LCIO and form them again. Then their momentum is 0, because it's filled by TrackState.getMomentum() which is not persisted. 
- If a situation like this happens (during my KF validation happened), then this value is 0. Track momentum is correct, tho

Reply:
Sounds like an hps-java bug to me, but I will comment it out. Is it safe to clobber the individual track momenta like you are saying?